File: TaskTracker.Model/User/Request/GetUserRequest.cs
using TaskTracker.Model.Attribute;
using TaskTracker.Model.Enum;

namespace TaskTracker.Model.User.Request
{
    public class GetUserRequest : PagedRequest
    {
        [CustomFilter(ComparisonType = ComparisonType.DoesNotContain, PropertyName = "Id")]
        public ICollection<long>? SkipIds { get; set; }

        [CustomFilter(ComparisonType = ComparisonType.StartsWith, PropertyName = "Email")]
        public string? SearchEmail { get; set; }

        [CustomFilter(ComparisonType = ComparisonType.Contains, PropertyName = "Id")]
        public ICollection<long>? Ids { get; set; }
    }
}

File: task-tracker-ui/src/app/model/user/GetUserRequest.ts
import PagedRequest, { mapPagedRequest } from "../PagedRequest";

export default interface GetUserRequest extends PagedRequest {
    skipIds?: number[],
    searchEmail?: string,
    ids?: number[]
}

export const mapGetUserRequest = (request: GetUserRequest) => {
    let params = mapPagedRequest(request)

    if(request.skipIds) {
        request.skipIds.forEach(id => params = params.append('SkipIds', id))
    }

    if(request.ids) {
        request.ids.forEach(id => params = params.append('Ids', id))
    }

    if(request.searchEmail) {
        params = params.append('SearchEmail', request.searchEmail)
    }

    return params
}